snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/EffectiveTimeRestorer.java

```diff
@@ -0,0 +1,503 @@
+				final String previousMapRule = (String) previousMember.getProperties().get(SnomedRf2Headers.FIELD_MAP_RULE);
+				final String previousMapAdvice = (String) previousMember.getProperties().get(SnomedRf2Headers.FIELD_MAP_ADVICE);
+				final String previousCorrelationId = (String) previousMember.getProperties().get(SnomedRf2Headers.FIELD_CORRELATION_ID);
+				final String previousMapCategoryId = (String) previousMember.getProperties().get(SnomedRf2Headers.FIELD_MAP_CATEGORY_ID);
+
+				if (previousMapTargetId != null && !previousMapTargetId.equals(memberToRestore.getMapTargetComponentId())) {
+					return false;
+				}
+
+				if (previousMapGroup != null && previousMapGroup.intValue() != memberToRestore.getMapGroup()) {
+					return false;
+				}
+
+				if (previousMapPriority != null && previousMapPriority.intValue() != memberToRestore.getMapPriority()) {
+					return false;
+				}
+
+				if (previousMapRule != null && !previousMapRule.equals(memberToRestore.getMapRule())) {
+					return false;
+				}
+
+				if (previousMapAdvice != null && !previousMapAdvice.equals(memberToRestore.getMapAdvice())) {
+					return false;
+				}
+
+				if (previousCorrelationId != null && !previousCorrelationId.equals(memberToRestore.getCorrelationId())) {
+					return false;
+				}
+				
+				// Handle extended map
+				if (memberToRestore.getMapCategoryId() != null && !memberToRestore.getMapCategoryId().equals(previousMapCategoryId)) {
+					return false;
+				}
+				
+				return memberToRestore.isActive() == previousMember.isActive() && memberToRestore.getModuleId().equals(previousMember.getModuleId());
+			} else if (componentToRestore instanceof SnomedSimpleMapRefSetMember) {
+				final SnomedSimpleMapRefSetMember memberToRestore = (SnomedSimpleMapRefSetMember) componentToRestore;
+				final String previousMapTargetDescription = (String) previousMember.getProperties().get(SnomedRf2Headers.FIELD_MAP_TARGET_DESCRIPTION);
+				
+				if (memberToRestore.getMapTargetComponentDescription() != null && !memberToRestore.getMapTargetComponentDescription().equals(previousMapTargetDescription)) {
+					return false;
+				}
+				
+				return memberToRestore.isActive() == previousMember.isActive()
+						&& memberToRestore.getModuleId().equals(previousMember.getModuleId()) 
+						&& memberToRestore.getMapTargetComponentId().equals(previousMember.getProperties().get(SnomedRf2Headers.FIELD_MAP_TARGET));
+						
+			}
+			if (componentToRestore.eClass().equals(SnomedRefSetPackage.Literals.SNOMED_REF_SET_MEMBER)) {
+				// Simple type reference set
+				final SnomedRefSetMember memberToRestore = (SnomedRefSetMember) componentToRestore;
+				return memberToRestore.isActive() == previousMember.isActive() && memberToRestore.getModuleId().equals(previousMember.getModuleId());
+			} else {
+				return false;
+			}
+		} else {
+			throw new UnexpectedTypeException("Unexpected member type '" + componentToRestore.getClass() + "'.");
+		}
```

Comment:
That's quite a long method but we are on a time schedule now, so I guess it should be ok.